go/vt/vtgate/engine/delete.go

```diff
@@ -190,6 +190,11 @@ func (del *Delete) GetFields(vcursor VCursor, bindVars map[string]*querypb.BindV
 	return nil, fmt.Errorf("BUG: unreachable code for %q", del.Query)
 }
 
+// Inputs is always empty for delete - it has no inputs.
+func (del *Delete) Inputs() []Primitive {
+	return []Primitive{}
```

Comment:
These can return `nil` instead. You could also make `Inputs` optional, and have only the primitives that have other inputs implement it. You verify by type asserting against an interface:
```
type HasInputs interface {
  Inputs() []Primitive
}
```
But required may actually be better eventhough it may lead to more boilerplate. At least, it will be explicit.